services/src/test/java/org/jfrog/artifactory/client/ArtifactoryTests.java

```diff
@@ -130,28 +134,22 @@ public void socketTimeoutBuilderTest() {
     @Test
     public void addInterceptorTest() {
```

Comment:
There seems be an issue with the new logic applied to this test. You are checking the value of the the *interceptor1Visits* variable before the interceptor had a chance to increment it, so the this test actually tests nothing right now.
The reason why the original code of the test (which seems to good) failed, is because a real bug, which was introduced by your code. The tests failed because this line **builder.build().restCall(req);** now throws an IOException, so the interceptors the test defines are never invoked. The reason why the restCall fails is because you're no longer allowing unauthenticated requests, which is s serious compatibility breakage.
The test code can be actually improved, so that it also fails if the restCall fails, with a clear error message.